components/context/src/main/java/datadog/context/EmptyContext.java

```diff
@@ -1,16 +1,24 @@
 package datadog.context;
 
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
 /** {@link Context} containing no values. */
+@ParametersAreNonnullByDefault
 final class EmptyContext implements Context {
   static final Context INSTANCE = new EmptyContext();
 
   @Override
+  @Nullable
   public <T> T get(ContextKey<T> key) {
     return null;
   }
 
   @Override
-  public <T> Context with(ContextKey<T> key, T value) {
+  public <T> Context with(ContextKey<T> key, @Nullable T value) {
+    if (key == null || value == null) {
```

Comment:
```suggestion
    if (value == null) {
```
Providing a `null` key is an error condition, so I'd like to keep it resulting in an exception.

Reply:
I will make the error explicit using `requireNonNull()` then (rather than getting an error on accessing the private key field on a null object).